lib/posthog/client.rb

```diff
@@ -342,10 +346,31 @@ def shutdown
 
     private
 
+    # before_send should run immediately before the event is sent to the queue.
+    # @param [Object] action The event to be sent to PostHog
+    # @return [null, Object, nil] The processed event or nil if the event should not be sent
+    def process_before_send(action)
+      return if action.nil? || action.empty?
+      return action unless @before_send
+
+      processed_action = @before_send.call(action)
+
+      if processed_action.nil?
+        logger.warn("Event #{action[:event]} was rejected in beforeSend function")
+      elsif processed_action.empty?
+        logger.warn("Event #{action[:event]} has no properties after beforeSend function, this is likely an error")
+      end
+
+      processed_action
+    end
+
     # private: Enqueues the action.
     #
     # returns Boolean of whether the item was added to the queue.
     def enqueue(action)
+      action = process_before_send(action)
+      return false if action.nil? || action.empty?
```

Comment:
technically this is a breaking change
but if someone was relying on the fact that we would enqueue empty objects then 🤷